engine/components-api/src/main/java/com/cloud/ha/HighAvailabilityManager.java

```diff
@@ -16,18 +16,36 @@
 // under the License.
 package com.cloud.ha;
 
-import java.util.List;
+import static java.lang.String.valueOf;
 
 import com.cloud.deploy.DeploymentPlanner;
 import com.cloud.host.HostVO;
 import com.cloud.host.Status;
 import com.cloud.utils.component.Manager;
 import com.cloud.vm.VMInstanceVO;
+import org.apache.cloudstack.framework.config.ConfigKey;
+
+import java.util.List;
 
 /**
  * HighAvailabilityManager checks to make sure the VMs are running fine.
  */
 public interface HighAvailabilityManager extends Manager {
+
+    ConfigKey<Integer> TimeBetweenCleanup = new ConfigKey<>("Advanced", Integer.class,
+        "time.between.cleanup", valueOf(3600 * 24), "Time in milliseconds to wait before the cleanup thread runs.",
+        false, null);
+
+    ConfigKey<Integer> MaxRetries = new ConfigKey<>("Advanced", Integer.class, "max.retries",
+        valueOf(5), "Number of times to retry start.", false, null);
+
+    ConfigKey<Integer> TimeToSleep = new ConfigKey<>("Advanced", Integer.class, "time.to.sleep",
+        valueOf(60 * 1000), "Time in seconds to sleep if no work items are found.", false, null);
+
+    ConfigKey<Integer> TimeBetweenFailures = new ConfigKey<>("Advanced", Integer.class,
+        "time.between.failures", valueOf(3600 * 1000), "Time in milliseconds before try to cleanup all the VMs"
+        + " which are registered for the HA event that were successful and are now ready to be purged.", false, null);
+
```

Comment:
the value should be "3600" seconds, and "seconds" in description.
Otherwise, you have to change line 855 in HighAvailabilityManagerImpl.java